hal/inc/ble_hal_defines.h

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (c) 2018 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef BLE_HAL_DEFINES_H
+#define BLE_HAL_DEFINES_H
+
+
+#if HAL_PLATFORM_BLE
+
+
+/**
+ * @addtogroup ble_defines
+ *
+ * @brief
+ *   This module provides BLE HAL layer macro definitions.
+ * @{
+ */
+
+/** @defgroup BLE device address
+ * @{ */
+#define BLE_SIG_ADDR_LEN                                        (6)
+
+// BLE device address type
+#define BLE_SIG_ADDR_TYPE_PUBLIC                                0x00 /**< Public (identity) address.*/
```

Comment:
Consider using enums throughout this file instead where appropriate as they provide better type safety.